cmd/monitor/config.go

```diff
@@ -16,23 +16,58 @@
 package main
 
 import (
+	"context"
+	"fmt"
 	"time"
 
 	"github.com/sigstore/rekor-monitor/pkg/identity"
 	"github.com/sigstore/rekor-monitor/pkg/notifications"
 )
 
 type IdentityMonitorConfiguration struct {
-	StartIndex                *int                                    `yaml:"startIndex"`
-	EndIndex                  *int                                    `yaml:"endIndex"`
-	MonitoredValues           identity.MonitoredValues                `yaml:"monitoredValues"`
-	ServerURL                 string                                  `yaml:"serverURL"`
-	OutputIdentitiesFile      string                                  `yaml:"outputIdentities"`
-	LogInfoFile               string                                  `yaml:"logInfoFile"`
-	IdentityMetadataFile      *string                                 `yaml:"identityMetadataFile"`
-	GitHubIssue               notifications.GitHubIssueInput          `yaml:"githubIssue"`
-	EmailNotificationSMTP     notifications.EmailNotificationInput    `yaml:"emailNotificationSMTP"`
-	EmailNotificationMailgun  notifications.MailgunNotificationInput  `yaml:"emailNotificationMailgun"`
-	EmailNotificationSendGrid notifications.SendGridNotificationInput `yaml:"emailNotificationSendGrid"`
-	Interval                  *time.Duration                          `yaml:"interval"`
+	StartIndex                *int                                     `yaml:"startIndex"`
+	EndIndex                  *int                                     `yaml:"endIndex"`
+	MonitoredValues           identity.MonitoredValues                 `yaml:"monitoredValues"`
+	ServerURL                 string                                   `yaml:"serverURL"`
+	OutputIdentitiesFile      string                                   `yaml:"outputIdentities"`
+	LogInfoFile               string                                   `yaml:"logInfoFile"`
+	IdentityMetadataFile      *string                                  `yaml:"identityMetadataFile"`
+	GitHubIssue               *notifications.GitHubIssueInput          `yaml:"githubIssue"`
+	EmailNotificationSMTP     *notifications.EmailNotificationInput    `yaml:"emailNotificationSMTP"`
+	EmailNotificationMailgun  *notifications.MailgunNotificationInput  `yaml:"emailNotificationMailgun"`
+	EmailNotificationSendGrid *notifications.SendGridNotificationInput `yaml:"emailNotificationSendGrid"`
+	Interval                  *time.Duration                           `yaml:"interval"`
+}
+
```

Comment:
What do you think about having a "pool" of notification providers? This would be splitting `TriggerNotifications` into two functions:

* Something like `CreateNotificationPlatformPool` which outputs `[]NotificationPlatform`, which you would populate based on what's set in the configuration
* `TriggerNotifications` which would iterate through the list and call `Send` on each

Functionally, these are equivalent, but it's a bit more idiomatic to use the interface.